langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Replace.java

```diff
@@ -43,16 +43,15 @@ private static BString replaceFromString(BRegexpValue regExp, BString str, BStri
                                              boolean isReplaceAll) {
         String originalStr = str.getValue();
         String replacementString = replacingStr.getValue();
-        Matcher matcher = RegexUtil.getMatcher(regExp, originalStr);
-        matcher.region(startIndex, originalStr.length());
+        String prefixStr = "";
+        String subStr = originalStr;
+        if (startIndex != 0) {
+            prefixStr = originalStr.substring(0, startIndex);
+            subStr = originalStr.substring(startIndex);
```

Comment:
```suggestion
            originalStr = originalStr.substring(startIndex);
```
and remove  variable `subStr`. WDYT?

Reply:
I don't see it as much of an improvement, but the suggestion will make it less readable. Whenever possible I try to avoid reassigning variables. Since these are string variables, the memory location will be the same. Hence, resolving